client/cypress/integration/query/parameter_spec.js

```diff
@@ -180,6 +180,69 @@ describe('Parameter', () => {
     });
   });
 
+  describe('Query Based Dropdown Parameter', () => {
+    beforeEach(() => {
+      const dropdownQueryData = {
+        name: 'Dropdown Query',
+        query: `SELECT 'value1' AS name, 1 AS value UNION ALL
+                SELECT 'value2' AS name, 2 AS value UNION ALL
+                SELECT 'value3' AS name, 3 AS value`,
+      };
+      createQuery(dropdownQueryData, true).then((dropdownQuery) => {
+        const queryData = {
+          name: 'Query Based Dropdown Parameter',
+          query: "SELECT '{{test-parameter}}' AS parameter",
+          options: {
+            parameters: [
+              { name: 'test-parameter',
+                title: 'Test Parameter',
+                type: 'query',
+                queryId: dropdownQuery.id },
+            ],
+          },
+        };
+
+        cy.visit(`/queries/${dropdownQuery.id}`);
+        cy.getByTestId('ExecuteButton').click();
+        cy.getByTestId('TableVisualization')
+          .should('contain', 'value1')
+          .and('contain', 'value2')
+          .and('contain', 'value3');
```

Comment:
Better move this to a test of its own? (I don't think setup is a right place for assertions)

Reply:
Considering Cypress style I don't see any problems on assertions in the setup (they are quite fast, new tests are not). But in this case I wanted to make sure the query was executed and had results (needed for the Dropdown), so the assertion is indeed part of the setup.